### DISCUSSION TITLE
/dashboard doesn't get automatically redirected to login

### DISCUSSION THREAD

**Discussion by wayneyaoo** [polymind-inc/acmebot#587] [Q&A] [answered]
Hi thanks for making the great app. I got a question about logging to the web page to manage certs. So I followed steps in "getting started" including the "enable authentication" part that enables AAD app as auth provider. But when I tried to login to `https://<functionanme>.azurewebsites.net/dashboard` or `https://<functionanme>.azurewebsites.net`, it merely returns 404 (I configured it to be 404 if unauthenticated) instead of redirecting me to login.

The created app was a new app registration. And I tried it on private mode and it doesn't work.

However, I peeked into the app and found the reply url to be `https://<functionanme>.azurewebsites.net/.auth/login/aad/callback`, and that's how I got a login redirection and successfully obtain an ID token to login.

So is this by design? Do I have to go to the reply url to login or I did something wrong?

**Comment by shibayan** [comment, marked as answer]
If you have a configuration that does not redirect to the login screen with a 302 if not authenticated, it would be a 404. You need to change the setting to redirect or access /.auth/aad/login manually.

**Reply by wayneyaoo** [reply]
Wait... you're saying the 301 redirection to "Microsoft" in the picture is redirecting to login page right?? omg I thought it was literally redirect to like microsoft.com... lol

**Reply by wayneyaoo** [reply]
Thank you! I fix it to 301 redirection. That works.
